build2.py

```diff
@@ -162,41 +213,60 @@ def make_pkg(package, dry=False, extra_args=''):
             print(e)
             raise
 
+def cli():
 
-if __name__ == "__main__":
-    import sys
-    import argparse
-    
     p = argparse.ArgumentParser()
-    build_pkgs = p.add_mutually_exclusive_group()
+    p.add_argument("path", default='.')
+    subp = p.add_subparsers(help="Build or split to make json of package "
+                                 "order/grouping. \n\tChoices: %(choices)s")
+    build_parser = subp.add_parser('build')
+    build_pkgs = build_parser.add_mutually_exclusive_group()
     build_pkgs.add_argument("-build", action='append', default=[])
     build_pkgs.add_argument("-buildall", action='store_true')
-    p.add_argument("-dry", action='store_true', default=False)
-    p.add_argument("-api", action='store_true', dest='recompile', default=False)
-    p.add_argument("-args", action='store', dest='cbargs', default='')
-    p.add_argument("-l", type=int, action='store', dest='level', default=0)
-    p.add_argument("-t", action='store_true', dest='t', default=False)
-    p.add_argument("-noautofail", action='store_false', dest='autofail', default=True)
-    p.add_argument("path", default='.')
+    build_pkgs.add_argument('-json-file-key', default=[], nargs=2)
+    build_parser.add_argument("-dry", action='store_true', default=False)
+    build_parser.add_argument("-api", action='store_true', dest='recompile', default=False)
+    build_parser.add_argument("-args", action='store', dest='cbargs', default='')
+    build_parser.add_argument("-l", type=int, action='store', dest='level', default=0)
+    build_parser.add_argument("-t", action='store_true', dest='t', default=False)
+    build_parser.add_argument("-noautofail", action='store_false', dest='autofail', default=True)
+    split_parser = subp.add_parser('split')
+    split_parser.add_argument('-t','--targetnum', type=int, default=10, help="How many packages in one anaconda build submission typically.")
```

Comment:
On this one, based on our discussions earlier, I think a more relevant parameter is target_time (in minutes).  This is fine for now, but perhaps let's work towards that in future PRs.